engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/ConfirmUrlPopup.java

```diff
@@ -73,21 +74,38 @@ public void setMessage(String title, String message) {
         }
     }
 
-    public void setCheckbox(WebBrowserConfig webBrowserConfig, String url) {
-        UICheckbox saveUrl = find("saveUrl", UICheckbox.class);
+    public void setCheckbox(WebBrowserConfig webBrowserConfig, String url) throws MalformedURLException {
+        UICheckbox saveHostName = find("saveHostName", UICheckbox.class);
 
-        if (saveUrl != null && webBrowserConfig != null) {
-            saveUrl.setChecked(false);
+        if (saveHostName != null) {
+            saveHostName.setChecked(false);
+            String hostName = new URL(url).getHost();
 
-            saveUrl.subscribe(checkbox -> {
-                boolean isTrustedUrl = saveUrl.isChecked();
+            saveHostName.subscribe(checkbox -> {
+                boolean isTrustedHostName = saveHostName.isChecked();
 
-                if (isTrustedUrl) {
-                    webBrowserConfig.addTrustedUrls(url);
+                if (isTrustedHostName) {
+                    webBrowserConfig.addTrustedHostName(hostName);
                 } else {
-                    webBrowserConfig.removeTrustedUrl(url);
+                    webBrowserConfig.removeTrustedHostName(hostName);
                 }
             });
+
+            UICheckbox saveUrl = find("saveUrl", UICheckbox.class);
+
+            if (saveUrl != null) {
```

Comment:
I highly doubt this should be nested in the `saveHostName` if. They should be separate imo

Reply:
Yes, they should be separate. Fixed!